actions/checkForAndPlaceOrder.ts

```diff
@@ -208,15 +208,13 @@ const _checkForAndPlaceOrder: ActionFn = async (
     }
   }
 
-  // TODO: Decide if we want to delete owners with no orders
-  //    - One one hand, we don't want to keep growing our registry forever
-  //    - On the other hand, it might be handy to know this owner has been verified before
-  // // Delete owners with no orders
-  // for (const [owner, conditionalOrders] of Array.from(ownerOrders.entries())) {
-  //   if (conditionalOrders.size === 0) {
-  //     ownerOrders.delete(owner);
-  //   }
-  // }
+  // It may be handy in other versions of the watch tower implemented in other languages
+  // (ie. not for Tenderly) to not delete owners, so we can keep track of them.
```

Comment:
Why would we want to keep track of former owners?
Is that for analytics purposes or something else?

Reply:
It can be for analytics as an example, but also for other optimisations. At the moment, when we add a contract, we retrieve it's code with `getCode` and check for the presence of some solidity functions within. If we leveraged an event watcher for `ChangedFallbackHandler`, and watched for changes on the `owners` we would have a way of optimising this code so that when existing owners add more conditional orders, we don't have to do this `getCode` again and save on some RPC calls.

Would require some benchmarking to see the effects on RPC call utilisation though.